src/main/java/net/imagej/ops/deconvolve/RichardsonLucyC.java

```diff
@@ -0,0 +1,231 @@
+			RichardsonLucyCorrection.class, RandomAccessibleInterval.class,
+			RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+			getFFTInput(), getFFTKernel());
+
+		fftIn = (UnaryComputerOp) Computers.unary(ops(), FFTMethodsOpC.class,
+			getFFTInput(), RandomAccessibleInterval.class);
+
+		fftKernel = (UnaryComputerOp) Computers.unary(ops(), FFTMethodsOpC.class,
+			getFFTKernel(), RandomAccessibleInterval.class);
+
+		copy = (UnaryHybridCF) Hybrids.unaryCF(ops(), Ops.Copy.RAI.class,
+			RandomAccessibleInterval.class, IntervalView.class);
+
+		copy2 = (UnaryHybridCF) Hybrids.unaryCF(ops(), Ops.Copy.RAI.class,
+			RandomAccessibleInterval.class, IntervalView.class);
+
+		create = (UnaryFunctionOp) Functions.unary(ops(), Ops.Create.Img.class,
+			Img.class, Dimensions.class, Util.getTypeFromInterval(out()));
+
+	}
+
+	@Override
+	public void performIterations(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		for (int i = 0; i < getMaxIterations(); i++) {
+
+			if (status != null) {
+				status.showProgress(i, getMaxIterations());
+			}
+
+			// create reblurred by convolving kernel with estimate
+			// NOTE: the FFT of the PSF of the kernel is performed in "preprocess" so
+			// no need to pass it in here.
+			convolver.compute1(this.getRAIExtendedEstimate(), this
+				.getRAIExtendedReblurred());
+
+			// compute correction factor
+			rlCorrection.compute2(in, getRAIExtendedReblurred(),
+				getRAIExtendedReblurred());
+
+			// perform update to calculate new estimate
+			update.compute1(getRAIExtendedReblurred(), getRAIExtendedEstimate());
+
+			// accelerate (take larger step)
+			if (getAccelerator() != null) {
+				getAccelerator().mutate(getRAIExtendedEstimate());
+			}
+		}
+
+		// crop padded back to original size
+		final long[] start = new long[out.numDimensions()];
+		final long[] end = new long[out.numDimensions()];
+
+		for (int d = 0; d < out.numDimensions(); d++) {
+			start[d] = 0;
+			end[d] = start[d] + out.dimension(d) - 1;
+		}
+
+		copy2.compute1(Views.interval(getRAIExtendedEstimate(), new FinalInterval(
+			start, end)), out);
+	}
+
+	@Override
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	protected void preProcess(RandomAccessibleInterval<I> in,
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		// if no output out of bounds factory exists create the obf for output
+		if (obfOutput == null) {
+			obfOutput =
+				new OutOfBoundsConstantValueFactory<O, RandomAccessibleInterval<O>>(Util
+					.getTypeFromInterval(out).createVariable());
+		}
+
+		// create image for the estimate, this image is defined over the entire
+		// convolution interval
+		final Img<O> estimate = create.compute1(getImgConvolutionInterval());
+
+		// create image for the reblurred
+		final Img<O> reblurred = create.compute1(getImgConvolutionInterval());
+
+		setRAIExtendedEstimate(estimate);
```

Comment:
addressed by af24a7025f5db191d2593a3516107025368cd5c1

that variable no longer belongs to the base class.  There is no longer a ```RichardsonLucyNonCirculantC```, instead there is only ```RichardsonLucyC```, and different implementations of RL can be achieved by composition.